packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts

```diff
@@ -29,14 +38,28 @@ export class SqsEventSource implements lambda.IEventSource {
   private _eventSourceMappingId?: string = undefined;
 
   constructor(readonly queue: sqs.IQueue, private readonly props: SqsEventSourceProps = { }) {
-    if (this.props.batchSize !== undefined && (this.props.batchSize < 1 || this.props.batchSize > 10)) {
-      throw new Error(`Maximum batch size must be between 1 and 10 inclusive (given ${this.props.batchSize})`);
+    if (this.props.maxBatchingWindow !== undefined && !Token.isUnresolved(this.props.maxBatchingWindow)) {
+      if (queue.fifo) {
+        throw new Error('Batching window is not supported for FIFO queues');
+      }
+      if (this.props.maxBatchingWindow.toSeconds() > 300) {
```

Comment:
These checks are still not precise. You will need to change it this way -

```suggestion
    if (this.props.maxBatchingWindow !== undefined) {
      if (queue.fifo) {
        throw new Error('Batching window is not supported for FIFO queues');
      }
      if (!this.props.maxBatchingWindow.isUnresolved() && this.props.maxBatchingWindow.toSeconds() > 300) {
```